saffron/src/read_proof.rs

```diff
@@ -0,0 +1,433 @@
+    data_comm: &Curve,
+) -> ReadProof
+where
+    RNG: RngCore + CryptoRng,
+{
+    let (_, endo_r) = Curve::endos();
+
+    let mut fq_sponge = CurveFqSponge::new(Curve::other_curve_sponge_params());
+
+    let data_d1 = Evaluations::from_vec_and_domain(data.to_vec(), domain.d1);
+    let data_poly: DensePolynomial<ScalarField> = data_d1.clone().interpolate();
+    let data_comm: PolyComm<Curve> = PolyComm {
+        chunks: vec![data_comm.clone()],
+    };
+
+    let query_d1 = Evaluations::from_vec_and_domain(query.to_vec(), domain.d1);
+    let query_poly: DensePolynomial<ScalarField> = query_d1.clone().interpolate();
+    let query_comm: PolyComm<Curve> = srs.commit_non_hiding(&query_poly, 1);
+
+    let answer_d1 = Evaluations::from_vec_and_domain(answer.to_vec(), domain.d1);
+    let answer_poly: DensePolynomial<ScalarField> = answer_d1.clone().interpolate();
+    let answer_comm: PolyComm<Curve> = srs.commit_non_hiding(&answer_poly, 1);
+
+    fq_sponge.absorb_g(&[
+        data_comm.chunks[0],
+        query_comm.chunks[0],
+        answer_comm.chunks[0],
+    ]);
+
+    // coefficient form, over d4? d2?
+    // quotient_Poly has degree d1
+    let quotient_poly: DensePolynomial<ScalarField> = {
+        // TODO: do not re-interpolate, we already did d1
+        let data_d2 = data_poly.evaluate_over_domain_by_ref(domain.d2);
+        let query_d2 = query_poly.evaluate_over_domain_by_ref(domain.d2);
+        let answer_d2 = answer_poly.evaluate_over_domain_by_ref(domain.d2);
+
+        // q×d - a
+        let numerator_eval: Evaluations<ScalarField, R2D<ScalarField>> =
+            &(&data_d2 * &query_d2) - &answer_d2;
+
+        let numerator_eval_interpolated = numerator_eval.interpolate();
+
+        let fail_final_q_division = || {
+            panic!("Division by vanishing poly must not fail at this point, we checked it before")
+        };
+        // We compute the polynomial t(X) by dividing the constraints polynomial
+        // by the vanishing polynomial, i.e. Z_H(X).
+        let (quotient, res) = numerator_eval_interpolated
+            .divide_by_vanishing_poly(domain.d1)
+            .unwrap_or_else(fail_final_q_division);
+        // As the constraints must be verified on H, the rest of the division
+        // must be equal to 0 as the constraints polynomial and Z_H(X) are both
+        // equal on H.
+        if !res.is_zero() {
+            fail_final_q_division();
+        }
+
+        quotient
+    };
+
+    // commit to the quotient polynomial $t$.
+    // num_chunks = 1 because our constraint is degree 2
+    let quotient_comm = srs.commit_non_hiding(&quotient_poly, 1).chunks[0];
+    fq_sponge.absorb_g(&[quotient_comm]);
+
+    // aka zeta
+    let evaluation_point = fq_sponge.squeeze(2);
+
+    // Fiat Shamir - absorbing evaluations
+    let fq_sponge_before_evaluations = fq_sponge.clone();
+    let mut fr_sponge = CurveFrSponge::new(Curve::sponge_params());
+    fr_sponge.absorb(&fq_sponge.digest());
```

Comment:
```suggestion
    let mut fr_sponge = CurveFrSponge::new(Curve::sponge_params());
    fr_sponge.absorb(&fq_sponge.clone().digest());
```
and use `fq_sponge` instead of `sq_sponge_before_evaluations`

Reply:
To me, it does not make sense to make a new variable as we could just use the old one, plus the name does not make sense to me.

Reply:
Ok